### PR TITLE
fix: remove syncthrough

### PR DESCRIPTION
**Pull request by ronag**
Inlining is both simpler and faster.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -183,11 +182,24 @@ class Client extends EventEmitter {
...
+          const onDrain = () => {
+            body.resume()
+          }
+
+          body.on('data', onData)
+          this.socket.on('drain', onDrain)
+
           this.socket.uncork()
           eos(body, (err) => {
+            body.destroy(err)
```
Comment (ronag) [inline lib/client.js:201]: Should probably also destroy the body here.

Comment (mcollina) [reply lib/client.js:202]: body is already finished here, so this is superfluous I think.
However, if the socket errors we should destroy the body. This is not done here.

Comment (ronag) [reply lib/client.js:202]: makes sense, I'll remove it.

**Comment by ronag** [comment]
We could do the same thing in the non chunked case and avoid `pipe()` with a possibly small memory & performance improvement, i.e. `pipe` does some not strictly necessary stuff.

**Comment by mcollina** [comment]
Send a PR, I’m happy to experiment with this module.

**Pull request merged**